### ISSUE TITLE
Error: Couldn't install: flake8-quotes 0.2.2

### ISSUE THREAD

**Issue by idgserpro** [zheller/flake8-quotes#19]
```bash
$ ./bin/buildout 
Develop: '/home/user/sites/brasil.gov.integracaocaldav/.'
Unused options for buildout: 'package-min-coverage'.
Updating instance.
Updating test.
Installing code-analysis.
Getting distribution for 'flake8-quotes'.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 1712, in main
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 1700, in with_ei_usage
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 1716, in <lambda>
  File "/home/user/pythons/python279/lib/python2.7/distutils/core.py", line 151, in setup
    dist.run_commands()
  File "/home/user/pythons/python279/lib/python2.7/distutils/dist.py", line 953, in run_commands
    self.run_command(cmd)
  File "/home/user/pythons/python279/lib/python2.7/distutils/dist.py", line 972, in run_command
    cmd_obj.run()
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 211, in run
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 427, in easy_install
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 476, in install_item
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 655, in install_eggs
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 930, in build_and_install
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/command/easy_install.py", line 919, in run_setup
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/sandbox.py", line 62, in run_setup
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/sandbox.py", line 105, in run
  File "/home/user/cache-python/eggs/setuptools-0.6c11-py2.7.egg/setuptools/sandbox.py", line 64, in <lambda>
  File "setup.py", line 3, in <module>
    from setuptools import find_packages
  File "/tmp/easy_install-7uVzSE/flake8-quotes-0.2.2/flake8_quotes.py", line 3, in <module>
ImportError: No module named pep8
An error occurred when trying to install flake8-quotes 0.2.2. Look above this message for any errors that were output by easy_install.
While:
  Installing code-analysis.
  Getting distribution for 'flake8-quotes'.
Error: Couldn't install: flake8-quotes 0.2.2
```

**Comment by idgserpro** [comment]
Is [this](https://github.com/zheller/flake8-quotes/commit/751dd8b829d7977dcf88b74e95cd42eb197d9639)?

**Comment by twolfson** [comment]
Ugh, it looks like the patch in #18 wasn't quite enough. I will attempt to resolve this by the end of the weekend but for now please lock down to an `0.1.x` version of `flake8-quotes`.

```bash
pip install flake8-quotes<0.2.0,>=0.1.2
```

**Comment by Ichimonji10** [comment]
I've faced this issue myself. You want to get at a version number from code, so you write the version number as code. But you also want to get the version number before that code is available, which creates a hen-and-egg packaging issue.

I've solved this issue in my own packages by placing the version number in a stand-alone file. The version number can then be got at in [`setup.py`](https://github.com/PulpQE/pulp-smash/blob/master/setup.py):

```python
with open('VERSION') as handle:
    VERSION = handle.read().strip()
```

This version number can also be gotten at in [`docs/conf.py`](https://github.com/PulpQE/pulp-smash/blob/master/docs/conf.py):

```python
# Parse the version according to PEP 440. An InvalidVersion exception is raised
# if the version is non-conformant, so the act of generating documentation
# serves as a unit test for the contents of the `VERSION` file.
with open(os.path.join(ROOT_DIR, 'VERSION')) as handle:
    VERSION = Version(handle.read().strip())
```

**Comment by twolfson** [comment]
Yea, I have seen that pattern before. We will probably take that approach since (iirc) `flake8` requires `__version__` to be defined. I usually keep it in the `setup.py` and call it a day =/

I would also like to see what other `flake8` packages do

**Comment by Ichimonji10** [comment]
You can set `__version__` from that file:

```python
with open('VERSION') as handle:
    __version__ = handle.read().strip()
```

That'll make `flake8` happy.

**Comment by Ichimonji10** [comment]
I'll be happy to submit a PR for this issue. It may take me a several hours to get to it, though, given I'm juggling other tasks.

**Comment by twolfson** [comment]
Even with a PR, I won't be able to land it until late today. Thanks for the thought though =)

I will have this resolved by the end of tonight (CST)

**Comment by Ichimonji10** [comment]
> Even with a PR, I won't be able to land it until late today.

No great rush. I'll pin dependencies in the meantime.

**Comment by Ichimonji10** [comment]
See #20.

**Comment by twolfson** [comment]
This should be resolved via #20 by @Ichimonji10 which was released in `0.2.3`.
